harmony/util.py

```diff
@@ -181,7 +186,8 @@ def bool_envvar(name: str, default: bool) -> bool:
         env=str_envvar('ENV', ''),
         text_logger=bool_envvar('TEXT_LOGGER', False),
         health_check_path=str_envvar('HEALTH_CHECK_PATH', '/tmp/health.txt'),
-        shared_secret_key=str_envvar('SHARED_SECRET_KEY', DEFAULT_SHARED_SECRET_KEY)
+        shared_secret_key=str_envvar('SHARED_SECRET_KEY', DEFAULT_SHARED_SECRET_KEY),
+        user_agent=str_envvar('USER_AGENT', 'harmony (unknown version)')
```

Comment:
I tested this branch out locally and logged the user agent. I'm seeing:
```
harmony (unknown version) harmony-service-lib/0.0.1
```
which makes sense if the USER_AGENT env var is not set. So my two questions are:
1.) What should we set the env var to in our deployments? Would it be something like?
`harmony/v0.0.0 (harmony-sit)` making the full user agent `harmony/v0.0.0 (harmony-sit) harmony-service-lib/0.0.1`
2.) Should we change the default to `harmony/unknown-version (unknown environment)` or does the current default look good?

Reply:
1. Ideally it'd be `harmony/actual-version-from-Bamboo (harmony-sit)`.  I think the full user agent looks correct-ish according to conventions, e.g. `Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0`
2. I prefer either the supplied default or moreover just making sure a sensible default is always set, e.g. a git commit shasum.

Reply:
1. In the Harmony project 767 branch I have it parsing the version from package.json so I think that should work. I should add parentheses though.
2. I think only older versions of harmony will not be supplying USER_AGENT, so maybe we can set a default harmony version as 0.0.0?
3. Do we actually need the optional 3rd user agent portion (which I added as an [optional param to "download"](https://github.com/nasa/harmony-service-lib-py/pull/10/files#diff-16bf995c8c9f397a140927d70a8119f947819ba2c48d6f35179248a6ad9e2683R269))? I recently realized that APP_NAME is passed to all of the argo workflows, meaning we could do something like `harmony/version (environment) harmony-service-lib/version **(service-name)**` rather than allowing the service provider to pass in something custom. I'm not sure what more someone would want to know about the user agent.

Reply:
Spoke with Patrick via Slack and we are going to use the app_name env var (as a user-agent comment) instead of allowing a custom service provider agent.